### DISCUSSION TITLE
gatsby-source-wordpress - allow `schema.perPage` and `schema.requestConcurrency` to be set by type

### DISCUSSION THREAD

**Discussion by tsdexter** [gatsbyjs/gatsby#35450] [Ideas / Feature Requests]
I have set my `schema` options like so: 

```
schema: {
          timeout: 1000000,
          perPage: 50,
          requestConcurrency: 10,
        },
```

I've limited these items because for pages and posts there seems to be a lot of work done to generate the result and the results can be huge (some pages are very content heavy). 

However, for other post types like tags, media item, category, menu, taxonomy, some custom post types, etc.. there is much less processing required and much smaller result sets that can likely be return 100, 200, maybe even 1000 at a time drastically reducing http requests.

I'd like to be able to set them like so: 

```
type: {
  MediaItem: {
         perPage: 100,
         requestConcurrency: 10,     
  },
  tag: {
         perPage: 500,
  },
```

**Comment by tsdexter** [comment]
@TylerBarnes  do you think this would be possible?

**Reply by TylerBarnes** [reply]
@tsdexter definitely. If you're interested in opening a PR for it I'll gladly review it! The logic should check for if there's a specific setting for that type and then fallback to the global setting if not. There's also a helper for getting settings for a specific type.

**Reply by tsdexter** [reply]
Great. I'll take a look when I have a bit of time. Thx
